src/ImmutableX.ts

```diff
@@ -96,116 +97,186 @@ export class ImmutableX {
    * @param ethSigner - the L1 signer
    * @param deposit - the token type amount in its corresponding unit
    * @returns the response of the resulting transaction
+   * @throws IMXError
    */
   public deposit(ethSigner: EthSigner, deposit: TokenAmount) {
-    return this.workflows.deposit(ethSigner, deposit);
+    return this.workflows
+      .deposit(ethSigner, deposit)
+      .then(res => res)
```

Comment:
Does `.then(res => res)` do anything? Seems unnecessary.

Reply:
If not required remove from other workflows too

Reply:
Not needed 👍

Reply:
https://github.com/immutable/imx-core-sdk/pull/172/commits/958ecc16a2bd2af4c7fd2bfd9defda71736e63d2